### ISSUE TITLE
Can OpenMP runtime be dropped?

### ISSUE THREAD

**Issue by makortel** [cerati/mkFit#257]
The mkfit workflow in CMSSW fails on ARM with
```
----- Begin Fatal Exception 19-Feb-2020 19:00:16 CET-----------------------
An exception of category 'PluginLibraryLoadError' occurred while
   [0] Constructing the EventProcessor
   [1] While attempting to load plugin MkFitOutputConverter
Exception Message:
unable to load /cvmfs/cms-ib.cern.ch/nweek-02616/slc7_aarch64_gcc820/cms/cmssw/CMSSW_11_1_X_2020-02-19-1100/lib/slc7_aarch64_gcc820/pluginRecoTrackerMkFitPlugins.so because /cvmfs/cms-ib.cern.ch/nweek-02616/slc7_aarch64_gcc820/external/gcc/8.3.0/lib64/libgomp.so.1: cannot allocate memory in static TLS block
----- End Fatal Exception -------------------------------------------------
```
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_aarch64_gcc820/CMSSW_11_1_X_2020-02-19-1100/pyRelValMatrixLogs/run/11634.7_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2021_GenSimFullINPUT+DigiFull_2021+RecoFull_trackingMkFit_2021+HARVESTFull_2021/step3_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2021_GenSimFullINPUT+DigiFull_2021+RecoFull_trackingMkFit_2021+HARVESTFull_2021.log#/

Trivial-looking solution would be to drop OpenMP use, but how that affects our current use of `#pragma omp simd`.

**Comment by srlantz** [comment]
We should probably be using -qopenmp-simd (Intel) and -fopenmp-simd (GNU), as these ignore all OpenMP pragmas except the simd ones, which are the only ones we are using. This might eliminate the dependency on the OpenMP runtimes.

The simd-only flags have been around for quite a while, since at least [icc 14.0](https://software.intel.com/sites/default/files/managed/37/76/RN_c_comp_140_initial_lin.pdf) and [gcc 4.9.0.](https://gcc.gnu.org/gcc-4.9/changes.html) In fact -qopenmp-simd is enabled by default in [icc 19.0 and up](https://software.intel.com/en-us/articles/a-survey-of-openmp-features-implemented-in-intel-fortran-and-c-compilers).

**Comment by srlantz** [comment]
@dan131riley pointed out that all the remaining OpenMP API calls, as well as OpenMP pragmas other than simd, are confined to just a few source files where they are hedged off with GPU/CUDA-related ifdef's (and in some cases with "#if 0"). So his suggestion (and I agree with it) is that we excise all the old GPU code, as it hasn't been maintained in a long time and at this point is incompatible with the rest of the code. It's still in the repository if we ever want to go back and look at it.

For the record, here's the full list of non-simd, non-commented-out references to OpenMP:

```
[1045] slantz@phi3 ~/mkFit-current$ grep -r '#pragma omp' | grep -v simd | \
> sed '/\/\/.*#pragma omp/d'

mkFit/FitterCU-imp.h:#pragma omp parallel for
mkFit/MkFitter.cc:#pragma omp parallel for private(itrack)
mkFit/MkFitter.cc:#pragma omp parallel for private(itrack)
mkFit/MkFitter.cc:#pragma omp parallel for private(itrack)
mkFit/fittestMPlex.cc:#pragma omp parallel reduction(+:s_tmp)

[1046] slantz@phi3 ~/mkFit-current$ grep -r 'omp_' | sed '/\/\/.*omp_/d'

mkFit/FitterCU-imp.h:    omp_set_num_threads(Config::numThreadsReorg);
mkFit/MkFitter.cc:  omp_set_num_threads(Config::numThreadsReorg);
mkFit/MkFitter.cc:  omp_set_num_threads(Config::numThreadsReorg);
mkFit/MkFitter.cc:  omp_set_num_threads(Config::numThreadsReorg);
mkFit/fittestMPlex.cc:  omp_set_nested(1);
mkFit/fittestMPlex.cc:  omp_set_num_threads(Config::numThreadsEvents);
mkFit/fittestMPlex.cc:  int numThreadsEvents = omp_get_num_threads();
mkFit/fittestMPlex.cc:  int thr_idx = omp_get_thread_num();
mkFit/fittestMPlex.cc:      if (omp_get_num_threads() <= 1) {

[1047] slantz@phi3 ~/mkFit-current$ grep -r '#include..omp\.h'

mkFit/FitterCU.h:#include <omp.h>
mkFit/fittestMPlex.cc:#include <omp.h>
```

**Comment by srlantz** [comment]
This is resolved by PR #279. The basic cure is to change the icc option to -qopenmp-simd and gcc option to -fopenmp-simd. To be extra safe, every line involving OpenMP directives (other than "omp simd") is now commented out.

PR #280 applies these same changes to the current CMS tagged version (V.2.0.1-0+pr241). The patched version is moving to its own branch as a result.

The additional goal of removing old GPU code is now tracked in a separate issue (#281).
